### PR TITLE
Cm/pse 7 rework

### PR DESCRIPTION
**Pull request by CheezItMan**
For the tests in the checkpoint, do we cover the "edge" cases (where you can't reshape the matrix into the given rows/columns as the question covers those examples.

I'm defaulting to covering them in the tests just because they are explicitly stated in the problem.  I suppose we could leave things more ambiguous by removing those from the question.

### CODE REVIEW

File: 07/reshape-a-matrix.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: paragraph
+* id: 931b7218-6fc5-4fd5-9b6c-7945856a6041
+* title: Ask Clarifying Questions
+* points: 3
+* topics: pse
```
Comment (beccaelenzil) [inline 07/reshape-a-matrix.md:87]:
```suggestion
* title: Ask Clarifying Questions
* topics: pse
```

File: 07/reshape-a-matrix.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+##### !placeholder
+
+Steps to solve a subproblem
+
+##### !end-placeholder
+
+<!-- other optional sections -->
+<!-- !hint - !end-hint (markdown, users can see after a failed attempt) -->
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+<!-- !explanation - !end-explanation (markdown, students can see after answering correctly) -->
```
Comment (beccaelenzil) [inline 07/reshape-a-matrix.md:276]:
```suggestion
### !explanation

Example Steps:
1. Store the number of rows and number of columns in the matrix in variables `row` and `column`
2. If `row`*`column` does not equal `r`*`c`, return the `matrix`
3. Initialize an empty list `reshaped_matrix`
4. Iterate through the number of rows
5. Initialize a new row 
6. Iterate through the columns
7. Append the correct item to the new row
8. Append the new row to the `reshaped_matrix`
9. return the `reshaped_matrix`

### !end-explanation
```

File: 07/reshape-a-matrix.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+## Reminders
+
+- Finish this assignment individually.
+- If you'd like a full list of our tips, hints, and notes, please bring up the document "[About Problem Solving Exercises](../about-pses/about-pses.md )" in unit.
```
Comment (beccaelenzil) [inline 07/reshape-a-matrix.md:285]:
```suggestion
```

**Comment by beccaelenzil** [comment]
I agree that we should include the edge test cases in the checkpoint since they are explicitly stated in the prompt. Good call!

My other suggestions are the conform with the format for PSEs 01-06.

Couple questions to think about:
- In the previous PSEs we don't include the test cases in pytest for the checkpoint since part of the PSE work is to write these test cases. What do you think about leaving them off here and in general? 
- We were thinking the only 0-3 graded question would be the final checkpoint question, that was instructors could provide critical feedback and then red/yellow/green grade the entire PSE. I realize I need to look back at all PSEs to make sure this was consistent.

**Comment by CheezItMan** [comment]
I like having the pytest versions of the tests just so students can understand what's being tested, but we could remove them with no problem, Leetocode doesn't let you see the tests either.  

I can change the points for the other questions, but if you leave off the point values it goes in as 0-1.  Would we also leave the big O questions without point values too?

**Comment by beccaelenzil** [comment]
I need to talk through the point values with you. Is there a way to make the entire assignment worth 3 points? Could with make points: 0? The thought is that instructors spend a lot of time assigning a point value to a single question, when we'd really like to think of the entire PSE holistically and assign it red/yellow/green. Similarly, students can place undo focus on a lost point or two.

**Comment by beccaelenzil** [comment]
With regards to showing the tests. I think including them is the right thing to do. We should add them for PSE 04 - 06. I think since they're in the checkpoint, students won't look at them before completing prompts 1-3

File: 07/reshape-a-matrix.checkpoint.md [hunk 4]
```diff
@@ -401,7 +208,58 @@ def reshape_matrix(matrix, r, c):
...
-- Finish this assignment individually.
-- If you'd like a full list of our tips, hints, and notes, please bring up the document "[About Problem Solving Exercises](../about-pses/about-pses.md )" in unit.
+### !end-challenge
+
+### !challenge
+
+* type: short-answer
+* id: 7bb7f05e-257c-43ba-a811-8900cfd622e9
+* title: What is the space complexity of your solution?
+* points: 0
```
Comment (beccaelenzil) [inline 07/reshape-a-matrix.checkpoint.md:244]: Talked to some instructors, and we actually decided 1 point it better. sorry for the back and forth!

```suggestion
* points: 1
```

File: 07/reshape-a-matrix.checkpoint.md [hunk 5]
```diff
@@ -401,7 +208,58 @@ def reshape_matrix(matrix, r, c):
...
 
-- Finish this assignment individually.
-- If you'd like a full list of our tips, hints, and notes, please bring up the document "[About Problem Solving Exercises](../about-pses/about-pses.md )" in unit.
+### !end-challenge
+
+### !challenge
+
+* type: short-answer
+* id: 7bb7f05e-257c-43ba-a811-8900cfd622e9
+* title: What is the space complexity of your solution?
```
Comment (beccaelenzil) [inline 07/reshape-a-matrix.checkpoint.md:243]:
```suggestion
* title: What is the space complexity of your solution? Explain.
```

File: 07/reshape-a-matrix.checkpoint.md [hunk 6]
```diff
@@ -401,7 +208,58 @@ def reshape_matrix(matrix, r, c):
...
+* points: 0
+* topics: Big-O, Python, lists, matrices
+
+##### !question
+
+What is the space complexity of your solution?
```
Comment (beccaelenzil) [inline 07/reshape-a-matrix.checkpoint.md:249]: Take it or leave it

```suggestion
What is the space complexity of your solution? Explain.
```

**Review by beccaelenzil — APPROVED** [review]

**Pull request merged**